File: data/generate.rb
#!/usr/bin/ruby
## Parses mime.types and creates mime.dat

file = File.read(ARGV.first || "mime.types")

table = {}
for line in file.lines
	next if line[0] == ?# or
		line.size <= 3 or
		!line.include?( ?\t )

	name, *extensions = line.split(/\s+/)
	for ext in extensions
		table[ext] = name
	end
end

File.open( 'mime.dat', 'w' ) do |f|
	f.write Marshal.dump( table )
end

File: ranger/data/generate.py
#!/usr/bin/python3
# coding=utf-8

if __name__ == '__main__':
	import sys, pickle

	protocol = 0
	table = {}

	for line in open(len(sys.argv) > 1 and sys.argv[1] or "mime.types"):
		if len(line) > 3 and line[0] != '#' and '\t' in line:
			name, *extensions = line.split()
			for ext in extensions:
				table[ext] = name

	pickle.dump(table, open('mime.dat', 'wb'), protocol)
